### PR TITLE
jbjark/early-return-if-not-supported-type

### PR DESCRIPTION
**Pull request by BjoerklundJacob**
Returns the node if not specific sparql builder for the type is supported by the connection

### CODE REVIEW

**Review by Hebbe1234 — APPROVED** [review]

File: VirtualAssistantBusinessLogic/VirtualAssistantBusinessLogic/KnowledgeGraph/KnowledgeGraph.cs [hunk 1]
```diff
@@ -26,9 +26,19 @@ public KnowledgeGraph(SparQLConnectionFactory sparQLConnectionFactory)
         public KnowledgeGraphNode FindNodeInformation(KnowledgeGraphNode node)
         {
             ISparQLConnection sparqlConnection = SparQLConnectionFactory.GetConnection();
-            ISparQLBuilder sparqlBuilder = sparqlConnection.GetSparQLBuilder(node.Information["Type"][0]);
+            // Get the types from the node that are supported by the connection
+            IEnumerable<string> supportedTypesFromNode = sparqlConnection.SupportedTypes.Intersect(node.Information["Type"]);
```
Comment (frejaz) [inline VirtualAssistantBusinessLogic/VirtualAssistantBusinessLogic/KnowledgeGraph/KnowledgeGraph.cs:30]: supportedTypesFromNode --> supportedTypesInNode

File: VirtualAssistantBusinessLogic/VirtualAssistantBusinessLogic/SparQL/WikidataSparQLConnection.cs [hunk 2]
```diff
@@ -19,9 +19,11 @@ public class WikidataSparQLConnection : ISparQLConnection
         public WikidataSparQLConnection()
         {
             ResponseDecoder = new XMLResponseDecoder();
+            SupportedTypes = new List<string> { "", "Human", "Country" };
```
Comment (frejaz) [inline VirtualAssistantBusinessLogic/VirtualAssistantBusinessLogic/SparQL/WikidataSparQLConnection.cs:22]: Add comment why "" is necessary

Comment (frejaz) [reply VirtualAssistantBusinessLogic/VirtualAssistantBusinessLogic/SparQL/WikidataSparQLConnection.cs:22]: Maybe mention that these types are from wikidata

**Review by frejaz — CHANGES_REQUESTED** [review]
smol things

**Review by Zzeeker — APPROVED** [review]

**Review by frejaz — APPROVED** [review]
nice

**Review by Zzeeker — APPROVED** [review]

**Pull request merged**